### PR TITLE
Fix MM46418

### PR DESCRIPTION
**Pull request by larkox**
#### Summary
https://github.com/mattermost/mattermost-mobile/pull/6578 added a workaround to fix an issue with scheme_admin. By looking at the code, the only possible issue I found is that we were observing the list and not observing with columns.

This should solve the issue.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-46418

#### Release Note
```release-note
Technical debt
```

### CODE REVIEW

**Comment by larkox** [comment]
@cpoile Can you verify this is indeed not resurfacing the issue you saw in the original PR? Thanks!

File: app/queries/servers/user.ts [hunk 1]
```diff
@@ -106,24 +104,17 @@ export const observeUserIsTeamAdmin = (database: Database, userId: string, teamI
     const id = `${teamId}-${userId}`;
     return database.get<TeamMembershipModel>(TEAM_MEMBERSHIP).query(
         Q.where('id', Q.eq(id)),
-    ).observe().pipe(
+    ).observeWithColumns(['scheme_admin']).pipe(
```
Comment (enahum) [inline app/queries/servers/user.ts:107]: should we add here distinctUntilChanged?

File: app/queries/servers/user.ts [hunk 2]
```diff
@@ -106,24 +104,17 @@ export const observeUserIsTeamAdmin = (database: Database, userId: string, teamI
...
         switchMap((tm) => of$(tm.length ? tm[0].schemeAdmin : false)),
     );
 };
 
 export const observeUserIsChannelAdmin = (database: Database, userId: string, channelId: string) => {
     const id = `${channelId}-${userId}`;
-    const myChannelRoles = observeMyChannel(database, channelId).pipe(
-        switchMap((mc) => of$(mc?.roles || '')),
-        distinctUntilChanged(),
-    );
```
Comment (enahum) [inline app/queries/servers/user.ts:119]: I think we still need to check the roles, don't we?

**Review by enahum — COMMENTED** [review]
couple of comments

File: app/queries/servers/user.ts (hunk 2 above)
Comment (larkox) [reply app/queries/servers/user.ts:119]: The roles was a workaround to deal with this value not being properly updated. That may also had to do with graphQL doing things wrong at that time.

I guess that 'scheme_admin' is up to date and will give you the same information as checking the roles, but 0/5.

**Comment by cpoile** [comment]
@larkox I'm not sure, you would have to go through the reproduction steps -- do you want me to do that? (It might have to be later in the week then.)

**Comment by larkox** [comment]
@cpoile I would prefer if you did, since you know the best what and how it failed. This doesn't seem to be a time critical issue, since it is just about removing the patch that makes sure it works, so I am ok waiting.

**Review by enahum — APPROVED** [review]

**Comment by enahum** [comment]
@cpoile can you please test this?

**Review by cpoile — APPROVED** [review]
I tested, and the channel permissions still work correctly with this PR. 👍 
However, I also reverted the original patch and tested, and I couldn't reproduce the original issue, so something must have changed in system to fix it.
Either way, this PR looks good.

**Pull request merged**